service/security/shiro/src/main/java/org/eclipse/kapua/service/authentication/shiro/AuthenticationServiceShiroImpl.java

```diff
@@ -111,6 +117,39 @@
     private final AccessPermissionService accessPermissionService = locator.getService(AccessPermissionService.class);
     private final AccessPermissionFactory accessPermissionFactory = locator.getFactory(AccessPermissionFactory.class);
 
+    static {
```

Comment:
We can no longer init shiro via a `static` into the `AuthenticatinServiceShiroImpl`.

Do you need this init otherwise test will not work properly?

Reply:
Yes, this is needed for the tests to work properly. Do you think there is any other way to introduce this configuration?

Reply:
Some time ago we ditched the usage of the `static` init in order to use the initialization via `shiro.ini` file. 
I think a `shiro.ini` file must be created under the `test/resources` directory of the test.

Reply:
Hi @zanpizmoht as @Coduz said static Shiro initialization was removed by #2905 
Since this qa tests are using the embedded environment the security context should be initialized by the broker (or the REST api, if any) during the startup and available for the test classes also.
If your test doesn't need embedded broker or REST api, so the Shiro context is not initialized, please use this approach: [cucumber Shiro init](https://github.com/eclipse/kapua/commit/c39b6856d2b3689784a68210a11428a3d1d81fd5#diff-0959952e566e7430b191a2c674b1352aR18)
The "Init Security Context" step does the Shiro context initilization and, at the end of the test, the step "Reset Security Context for all scenarios" reset the Shiro context.
There is a probably not needed shiro.ini file in the qa path (qa-common) and @lorthirk is working on it.
So please, remove the static initialization and try adding the step I suggested to your test and give me a feedback.

Reply:
I can confirm that the `shiro.ini` under `kapua-qa/common/src/main/resources` is not needed. In [this branch](https://github.com/lorthirk/kapua/tree/change-removeQaCommonShiroIni) I attempted to remove it, and as you can see from [Travis](https://travis-ci.com/github/lorthirk/kapua/builds/167098920) the build was successful.

Reply:
I have fixed this issue by removing static configuration and added shiro configuration to the Cucumber test scenarios